### PR TITLE
ceph: fix error in <ceph orch device ls> test

### LINKED ISSUE #8530: CephMgrSuite is failing to list devices
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
The mgr test is failing intermittently with an error such as this:

```
2021-08-12 09:17:00.369920 D | exec: Running command: kubectl exec -i rook-ceph-tools-78cdfd976c-dp5cz -n mgr-ns -- timeout 15 ceph orch device ls --connect-timeout=15
2021-08-12 09:17:00.879561 W | installer: Error executing command "ceph": <exit status 22>
    ceph_mgr_test.go:185: 
        	Error Trace:	ceph_mgr_test.go:185
        	Error:      	Expected nil, but got: &exec.ExitError{ProcessState:(*os.ProcessState)(0xc00013b1a0), Stderr:[]uint8{0x45, 0x72, 0x72, 0x6f, 0x72, 0x20, 0x45, 0x49, 0x4e, 0x56, 0x41, 0x4c, 0x3a, 0x20, 0x54, 0x72, 0x61, 0x63, 0x65, 0x62, 0x61, 0x63, 0x6b, 0x20, 0x28, 0x6d, 0x6f, 0x73, 0x74, 0x20, 0x72, 0x65, 0x63, 0x65, 0x6e, 0x74, 0x20, 0x63, 0x61, 0x6c, 0x6c, 0x20, 0x6c, 0x61, 0x73, 0x74, 0x29, 0x3a, 0xa, 0x20, 0x20, 0x46, 0x69, 0x6c, 0x65, 0x20, 0x22, 0x2f, 0x75, 0x73, 0x72, 0x2f, 0x73, 0x68, 0x61, 0x72, 0x65, 0x2f, 0x63, 0x65, 0x70, 0x68, 0x2f, 0x6d, 0x67, 0x72, 0x2f, 0x6d, 0x67, 0x72, 0x5f, 0x6d, 0x6f, 0x64, 0x75, 0x6c, 0x65, 0x2e, 0x70, 0x79, 0x22, 0x2c, 0x20, 0x6c, 0x69, 0x6e, 0x65, 0x20, 0x31, 0x35, 0x36, 0x33, 0x2c, 0x20, 0x69, 0x6e, 0x20, 0x5f, 0x68, 0x61, 0x6e, 0x64, 0x6c, 0x65, 0x5f, 0x63, 0x6f, 0x6d, 0x6d, 0x61, 0x6e, 0x64, 0xa, 0x20, 0x20, 0x20, 0x20, 0x72, 0x65, 0x74, 0x75, 0x72, 0x6e, 0x20, 0x73, 0x65, 0x6c, 0x66, 0x2e, 0x68, 0x61, 0x6e, 0x64, 0x6c, 0x65, 0x5f, 0x63, 0x6f, 0x6d, 0x6d, 0x61, 0x6e, 0x64, 0x28, 0x69, 0x6e, 0x62, 0x75, 0x66, 0x2c, 0x20, 0x63, 0x6d, 0x64, 0x29, 0xa, 0x20, 0x20, 0x46, 0x69, 0x6c, 0x65, 0x20, 0x22, 0x2f, 0x75, 0x73, 0x72, 0x2f, 0x73, 0x68, 0x61, 0x72, 0x65, 0x2f, 0x63, 0x65, 0x70, 0x68, 0x2f, 0x6d, 0x67, 0x72, 0x2f, 0x6f, 0x72, 0x63, 0x68, 0x65, 0x73, 0x74, 0x72, 0x61, 0x74, 0x6f, 0x72, 0x2f, 0x5f, 0x69, 0x6e, 0x74, 0x65, 0x72, 0x66, 0x61, 0x63, 0x65, 0x2e, 0x70, 0x79, 0x22, 0x2c, 0x20, 0x6c, 0x69, 0x6e, 0x65, 0x20, 0x31, 0x36, 0x37, 0x2c, 0x20, 0x69, 0x6e, 0x20, 0x68, 0x61, 0x6e, 0x64, 0x6c, 0x65, 0x5f, 0x63, 0x6f, 0x6d, 0x6d, 0x61, 0x6e, 0x64, 0xa, 0x20, 0x20, 0x20, 0x20, 0x72, 0x65, 0x74, 0x75, 0x72, 0x6e, 0x20, 0x64, 0x69, 0x73, 0x70, 0x61, 0x74, 0x63, 0x68, 0x5b, 0x63, 0x6d, 0x64, 0x5b, 0x27, 0x70, 0x72, 0x65, 0x66, 0x69, 0x78, 0x27, 0x5d, 0x5d, 0x2e, 0x63, 0x61, 0x6c, 0x6c, 0x28, 0x73, 0x65, 0x6c, 0x66, 0x2c, 0x20, 0x63, 0x6d, 0x64, 0x2c, 0x20, 0x69, 0x6e, 0x62, 0x75, 0x66, 0x29, 0xa, 0x20, 0x20, 0x46, 0x69, 0x6c, 0x65, 0x20, 0x22, 0x2f, 0x75, 0x73, 0x72, 0x2f, 0x73, 0x68, 0x61, 0x72, 0x65, 0x2f, 0x63, 0x65, 0x70, 0x68, 0x2f, 0x6d, 0x67, 0x72, 0x2f, 0x6d, 0x67, 0x72, 0x5f, 0x6d, 0x6f, 0x64, 0x75, 0x6c, 0x65, 0x2e, 0x70, 0x79, 0x22, 0x2c, 0x20, 0x6c, 0x69, 0x6e, 0x65, 0x20, 0x34, 0x31, 0x34, 0x2c, 0x20, 0x69, 0x6e, 0x20, 0x63, 0x61, 0x6c, 0x6c, 0xa, 0x20, 0x20, 0x20, 0x20, 0x72, 0x65, 0x74, 0x75, 0x72, 0x6e, 0x20, 0x73, 0x65, 0x6c, 0x66, 0x2e, 0x66, 0x75, 0x6e, 0x63, 0x28, 0x6d, 0x67, 0x72, 0x2c, 0x20, 0x2a, 0x2a, 0x6b, 0x77, 0x61, 0x72, 0x67, 0x73, 0x29, 0xa, 0x20, 0x20, 0x46, 0x69, 0x6c, 0x65, 0x20, 0x22, 0x2f, 0x75, 0x73, 0x72, 0x2f, 0x73, 0x68, 0x61, 0x72, 0x65, 0x2f, 0x63, 0x65, 0x70, 0x68, 0x2f, 0x6d, 0x67, 0x72, 0x2f, 0x6f, 0x72, 0x63, 0x68, 0x65, 0x73, 0x74, 0x72, 0x61, 0x74, 0x6f, 0x72, 0x2f, 0x5f, 0x69, 0x6e, 0x74, 0x65, 0x72, 0x66, 0x61, 0x63, 0x65, 0x2e, 0x70, 0x79, 0x22, 0x2c, 0x20, 0x6c, 0x69, 0x6e, 0x65, 0x20, 0x31, 0x30, 0x37, 0x2c, 0x20, 0x69, 0x6e, 0x20, 0x3c, 0x6c, 0x61, 0x6d, 0x62, 0x64, 0x61, 0x3e, 0xa, 0x20, 0x20, 0x20, 0x20, 0x77, 0x72, 0x61, 0x70, 0x70, 0x65, 0x72, 0x5f, 0x63, 0x6f, 0x70, 0x79, 0x20, 0x3d, 0x20, 0x6c, 0x61, 0x6d, 0x62, 0x64, 0x61, 0x20, 0x2a, 0x6c, 0x5f, 0x61, 0x72, 0x67, 0x73, 0x2c, 0x20, 0x2a, 0x2a, 0x6c, 0x5f, 0x6b, 0x77, 0x61, 0x72, 0x67, 0x73, 0x3a, 0x20, 0x77, 0x72, 0x61, 0x70, 0x70, 0x65, 0x72, 0x28, 0x2a, 0x6c, 0x5f, 0x61, 0x72, 0x67, 0x73, 0x2c, 0x20, 0x2a, 0x2a, 0x6c, 0x5f, 0x6b, 0x77, 0x61, 0x72, 0x67, 0x73, 0x29, 0x20, 0x20, 0x23, 0x20, 0x6e, 0x6f, 0x71, 0x61, 0x3a, 0x20, 0x45, 0x37, 0x33, 0x31, 0xa, 0x20, 0x20, 0x46, 0x69, 0x6c, 0x65, 0x20, 0x22, 0x2f, 0x75, 0x73, 0x72, 0x2f, 0x73, 0x68, 0x61, 0x72, 0x65, 0x2f, 0x63, 0x65, 0x70, 0x68, 0x2f, 0x6d, 0x67, 0x72, 0x2f, 0x6f, 0x72, 0x63, 0x68, 0x65, 0x73, 0x74, 0x72, 0x61, 0x74, 0x6f, 0x72, 0x2f, 0x5f, 0x69, 0x6e, 0x74, 0x65, 0x72, 0x66, 0x61, 0x63, 0x65, 0x2e, 0x70, 0x79, 0x22, 0x2c, 0x20, 0x6c, 0x69, 0x6e, 0x65, 0x20, 0x39, 0x36, 0x2c, 0x20, 0x69, 0x6e, 0x20, 0x77, 0x72, 0x61, 0x70, 0x70, 0x65, 0x72, 0xa, 0x20, 0x20, 0x20, 0x20, 0x72, 0x65, 0x74, 0x75, 0x72, 0x6e, 0x20, 0x66, 0x75, 0x6e, 0x63, 0x28, 0x2a, 0x61, 0x72, 0x67, 0x73, 0x2c, 0x20, 0x2a, 0x2a, 0x6b, 0x77, 0x61, 0x72, 0x67, 0x73, 0x29, 0xa, 0x20, 0x20, 0x46, 0x69, 0x6c, 0x65, 0x20, 0x22, 0x2f, 0x75, 0x73, 0x72, 0x2f, 0x73, 0x68, 0x61, 0x72, 0x65, 0x2f, 0x63, 0x65, 0x70, 0x68, 0x2f, 0x6d, 0x67, 0x72, 0x2f, 0x6f, 0x72, 0x63, 0x68, 0x65, 0x73, 0x74, 0x72, 0x61, 0x74, 0x6f, 0x72, 0x2f, 0x6d, 0x6f, 0x64, 0x75, 0x6c, 0x65, 0x2e, 0x70, 0x79, 0x22, 0x2c, 0x20, 0x6c, 0x69, 0x6e, 0x65, 0x20, 0x34, 0x35, 0x31, 0x2c, 0x20, 0x69, 0x6e, 0x20, 0x5f, 0x6c, 0x69, 0x73, 0x74, 0x5f, 0x64, 0x65, 0x76, 0x69, 0x63, 0x65, 0x73, 0xa, 0x20, 0x20, 0x20, 0x20, 0x69, 0x6e, 0x76, 0x5f, 0x68, 0x6f, 0x73, 0x74, 0x73, 0x20, 0x3d, 0x20, 0x72, 0x61, 0x69, 0x73, 0x65, 0x5f, 0x69, 0x66, 0x5f, 0x65, 0x78, 0x63, 0x65, 0x70, 0x74, 0x69, 0x6f, 0x6e, 0x28, 0x63, 0x6f, 0x6d, 0x70, 0x6c, 0x65, 0x74, 0x69, 0x6f, 0x6e, 0x29, 0xa, 0x20, 0x20, 0x46, 0x69, 0x6c, 0x65, 0x20, 0x22, 0x2f, 0x75, 0x73, 0x72, 0x2f, 0x73, 0x68, 0x61, 0x72, 0x65, 0x2f, 0x63, 0x65, 0x70, 0x68, 0x2f, 0x6d, 0x67, 0x72, 0x2f, 0x6f, 0x72, 0x63, 0x68, 0x65, 0x73, 0x74, 0x72, 0x61, 0x74, 0x6f, 0x72, 0x2f, 0x5f, 0x69, 0x6e, 0x74, 0x65, 0x72, 0x66, 0x61, 0x63, 0x65, 0x2e, 0x70, 0x79, 0x22, 0x2c, 0x20, 0x6c, 0x69, 0x6e, 0x65, 0x20, 0x32, 0x32, 0x34, 0x2c, 0x20, 0x69, 0x6e, 0x20, 0x72, 0x61, 0x69, 0x73, 0x65, 0x5f, 0x69, 0x66, 0x5f, 0x65, 0x78, 0x63, 0x65, 0x70, 0x74, 0x69, 0x6f, 0x6e, 0xa, 0x20, 0x20, 0x20, 0x20, 0x72, 0x61, 0x69, 0x73, 0x65, 0x20, 0x65, 0xa, 0x45, 0x78, 0x63, 0x65, 0x70, 0x74, 0x69, 0x6f, 0x6e, 0x3a, 0x20, 0x4e, 0x6f, 0x20, 0x73, 0x74, 0x6f, 0x72, 0x61, 0x67, 0x65, 0x20, 0x63, 0x6c, 0x61, 0x73, 0x73, 0x20, 0x65, 0x78, 0x69, 0x73, 0x74, 0x73, 0x20, 0x6d, 0x61, 0x74, 0x63, 0x68, 0x69, 0x6e, 0x67, 0x20, 0x6e, 0x61, 0x6d, 0x65, 0x20, 0x70, 0x72, 0x6f, 0x76, 0x69, 0x64, 0x65, 0x64, 0x20, 0x69, 0x6e, 0x20, 0x63, 0x65, 0x70, 0x68, 0x20, 0x63, 0x6f, 0x6e, 0x66, 0x69, 0x67, 0x20, 0x61, 0x74, 0x20, 0x6d, 0x67, 0x72, 0x2f, 0x72, 0x6f, 0x6f, 0x6b, 0x2f, 0x73, 0x74, 0x6f, 0x72, 0x61, 0x67, 0x65, 0x5f, 0x63, 0x6c, 0x61, 0x73, 0x73, 0xa, 0xa, 0x63, 0x6f, 0x6d, 0x6d, 0x61, 0x6e, 0x64, 0x20, 0x74, 0x65, 0x72, 0x6d, 0x69, 0x6e, 0x61, 0x74, 0x65, 0x64, 0x20, 0x77, 0x69, 0x74, 0x68, 0x20, 0x65, 0x78, 0x69, 0x74, 0x20, 0x63, 0x6f, 0x64, 0x65, 0x20, 0x32, 0x32, 0xa}}
        	Test:       	TestCephMgrSuite/TestDeviceLs
2021-08-12 09:17:00.879857 I | integrationTest: output = . Error EINVAL: Traceback (most recent call last):
  File "/usr/share/ceph/mgr/mgr_module.py", line 1563, in _handle_command
    return self.handle_command(inbuf, cmd)
  File "/usr/share/ceph/mgr/orchestrator/_interface.py", line 167, in handle_command
    return dispatch[cmd['prefix']].call(self, cmd, inbuf)
  File "/usr/share/ceph/mgr/mgr_module.py", line 414, in call
    return self.func(mgr, **kwargs)
  File "/usr/share/ceph/mgr/orchestrator/_interface.py", line 107, in <lambda>
    wrapper_copy = lambda *l_args, **l_kwargs: wrapper(*l_args, **l_kwargs)  # noqa: E731
  File "/usr/share/ceph/mgr/orchestrator/_interface.py", line 96, in wrapper
    return func(*args, **kwargs)
  File "/usr/share/ceph/mgr/orchestrator/module.py", line 451, in _list_devices
    inv_hosts = raise_if_exception(completion)
  File "/usr/share/ceph/mgr/orchestrator/_interface.py", line 224, in raise_if_exception
    raise e
Exception: No storage class exists matching name provided in ceph config at mgr/rook/storage_class

command terminated with exit code 22
```

See the full log [here](https://github.com/rook/rook/pull/8211/checks?check_run_id=3309877450).

**Expected behavior:**
Integration tests should pass consistently.

### PR DESCRIPTION
**Pull request by jmolmo**
Rook orchestrator is going to use PVs as _devices_ for OSDs.

These PVs must use one storage class present in the cluster and configured properly in the manager.
With this change we include this new requirement in the Rook manager module integration test.

Signed-off-by: Juan Miguel Olmo Martínez <jolmomar@redhat.com>

**Description of your changes:**

Resolves #8530

### CODE REVIEW

File: tests/integration/ceph_mgr_test.go [hunk 1]
```diff
@@ -115,6 +116,30 @@ func (s *CephMgrSuite) execute(command []string) (error, string) {
...
+metadata:
+  name: ` + storageClassName + `
+provisioner: kubernetes.io/no-provisioner
+volumeBindingMode: WaitForFirstConsumer
+`
+	err := s.k8sh.ResourceOperation("apply", localStorageClass)
+	if err == nil {
+		err, _ = s.installer.Execute("ceph", []string{"config", "set", "mgr", "mgr/rook/storage_class", storageClassName}, s.namespace)
+		if err == nil {
+			logger.Infof("Storage class <%q> set in manager config ", storageClassName)
```
Comment (travisn) [inline tests/integration/ceph_mgr_test.go:134]:
```suggestion
			logger.Infof("Storage class %q set in manager config ", storageClassName)
```

File: tests/integration/ceph_mgr_test.go [hunk 2]
```diff
@@ -115,6 +116,30 @@ func (s *CephMgrSuite) execute(command []string) (error, string) {
...
+		} else {
+			logger.Errorf("Error configuring local storage class in manager config: %q", err)
```
Comment (travisn) [inline tests/integration/ceph_mgr_test.go:136]:
```suggestion
			assert.Fail(s.T(), "Error configuring local storage class in manager config: %q", err)
```

File: tests/integration/ceph_mgr_test.go [hunk 3]
```diff
@@ -115,6 +116,30 @@ func (s *CephMgrSuite) execute(command []string) (error, string) {
...
+		}
+	} else {
+		logger.Errorf("Error creating local storage class: %q ", err)
```
Comment (travisn) [inline tests/integration/ceph_mgr_test.go:139]:
```suggestion
		assert.Fail(s.T(), "Error creating local storage class: %q ", err)
```

**Review by travisn — CHANGES_REQUESTED** [review]
The device test is fixed, but now the status test is still failing:

```
=== RUN   TestCephMgrSuite/TestStatus
2021-08-17 10:30:13.749964 I | integrationTest: Testing .... <ceph orch status>
2021-08-17 10:30:13.749985 D | exec: Running command: kubectl exec -i rook-ceph-tools-78cdfd976c-hbxp6 -n mgr-ns -- timeout 15 ceph orch status --connect-timeout=15
2021-08-17 10:30:28.981536 W | installer: Error executing command "ceph": <exit status 124>
    ceph_mgr_test.go:217: 
        	Error Trace:	ceph_mgr_test.go:217
        	Error:      	Expected nil, but got: &exec.ExitError{ProcessState:(*os.ProcessState)(0xc0004c1248), Stderr:[]uint8{0x63, 0x6f, 0x6d, 0x6d, 0x61, 0x6e, 0x64, 0x20, 0x74, 0x65, 0x72, 0x6d, 0x69, 0x6e, 0x61, 0x74, 0x65, 0x64, 0x20, 0x77, 0x69, 0x74, 0x68, 0x20, 0x65, 0x78, 0x69, 0x74, 0x20, 0x63, 0x6f, 0x64, 0x65, 0x20, 0x31, 0x32, 0x34, 0xa}}
        	Test:       	TestCephMgrSuite/TestStatus
2021-08-17 10:30:28.981949 I | integrationTest: output = . command terminated with exit code 124
    ceph_mgr_test.go:220: 
        	Error Trace:	ceph_mgr_test.go:220
        	Error:      	Not equal: 
        	            	expected: ". command terminated with exit code 124"
        	            	actual  : "Backend: rook\nAvailable: Yes"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1,2 @@
        	            	-. command terminated with exit code 124
        	            	+Backend: rook
        	            	+Available: Yes
        	Test:       	TestCephMgrSuite/TestStatus
```

**Comment by jmolmo** [comment]
> The device test is fixed, but now the status test is still failing:
> 
> ```
> === RUN   TestCephMgrSuite/TestStatus
> ...
> 2021-08-17 10:30:28.981949 I | integrationTest: output = . command terminated with exit code 124

> ```

I have executed the test 20 times, and this error appears only in a couple of times .. in different tests. I need to investigate more... but it seems that "ceph" commands are not working in random moments.. i haven't found any error in mgr/mon logs ... so for the moment no idea about what can be the root cause.

**Comment by travisn** [comment]
> I have executed the test 20 times, and this error appears only in a couple of times .. in different tests. I need to investigate more... but it seems that "ceph" commands are not working in random moments.. i haven't found any error in mgr/mon logs ... so for the moment no idea about what can be the root cause.

Ok let's go ahead and merge this and we can investigate the ceph command issue separately, but at least it is not severe if the frequency is that low.

**Review by travisn — APPROVED** [review]

**Pull request merged**